### ISSUE TITLE
Apply to customized dataset

### ISSUE THREAD

**Issue by ding3820** [TungChintao/APM#2]
Hi,

I appreciate your insightful work! The proposed method is both elegant and effective. I’m currently exploring its application to my own dataset but have a few questions regarding the implementation. I would greatly appreciate any guidance you can provide.

- Did you train your own version of HSNet, or did you use the released weights? If possible, would you be willing to share your checkpoint?
- What is the recommended approach for applying your method to a new model with a customized dataset? From my understanding, the workflow might be:
    1.  Load the pretrained weights of the entire model
    2. Froze the model and train APM and ACPA on the customized dataset
    3. Perform inference on the customized dataset

Could you confirm if this is correct or suggest any necessary modifications?

Thank you in advance for your time and help!

**Comment by TungChintao** [comment]
Thank you for your interest in our work. 

1) We trained HSNet on the source domain (Pascal dataset) following PATNet's configuration. For detailed training methods, please referring to the [PATNet](https://github.com/slei109/PATNet) and [HSNet](https://github.com/juhongm999/hsnet). Our checkpoint trained on the source domain is available here: https://drive.google.com/file/d/1qvsMneFbWyZoaux45TiD-Rvucew1EK4_/view?usp=drive_link

2) Your understanding of the workflow is absolutely correct: (i) loading pretrained weights, freezing the base model; (ii) fine-tuning the adaptation modules (APM and ACPA) on your custom dataset, and then (iii) performing inference.

Please let me know if you have any further questions about implementation details.

**Comment by Yongze** [comment]
Hi,
Based on your response above, I have questiones on the workflow (ii) and (iii):

1. The paper in Sec 4.2 mentioned that the method uses labels of support set to fine-tune AMP and ACPA through CE loss. This makes me confused. According to the setting of FSS, we use n-shot support image/label to predict the query mask. But, now, the method predicts support mask and use support label to calculate the loss in adaptation stage. Does that mean you are doing fully supervised learning on target domain dataset in this stage.
2. If the model is fine-tuned in target domain dataset in adaptation stage, will the model be evaluated in the same dataset? Or did you use 5-fold ?

Could you confirm if this is correct or suggest any necessary modifications?

Thank you in advance for your time and help!

**Comment by TungChintao** [comment]
> Hi, Based on your response above, I have questiones on the workflow (ii) and (iii):
> 
> 1. The paper in Sec 4.2 mentioned that the method uses labels of support set to fine-tune AMP and ACPA through CE loss. This makes me confused. According to the setting of FSS, we use n-shot support image/label to predict the query mask. But, now, the method predicts support mask and use support label to calculate the loss in adaptation stage. Does that mean you are doing fully supervised learning on target domain dataset in this stage.
> 2. If the model is fine-tuned in target domain dataset in adaptation stage, would you use 5-fold to evaluate the fine-tuned model on the same dataset?
> 
> Could you confirm if this is correct or suggest any necessary modifications?
> 
> Thank you in advance for your time and help!

Thank you for your interest in our work.
Our fine-tuning process only utilizes the support set, without involving the query set, while still maintaining the n-way k-shot setting during the fine-tuning phase. In other words, the fine-tuning is task-based.
